src/main/java/com/gtnewhorizon/gtnhlib/CommonProxy.java

```diff
@@ -86,4 +92,11 @@ public void sendMessageAboveHotbar(EntityPlayerMP player, IChatComponent chatCom
         NetworkHandler.instance
                 .sendTo(new PacketMessageAboveHotbar(chatComponent, displayDuration, drawShadow, shouldFade), player);
     }
+
+    @SubscribeEvent
+    public static void onPlayerLogin(PlayerEvent.PlayerLoggedInEvent event) {
+        if (!(event.player instanceof EntityPlayerMP playerMP)) return;
+        int distance = MinecraftServer.getServer().getConfigurationManager().getViewDistance();
+        NetworkHandler.instance.sendTo(new PacketViewDistance(distance), playerMP);
+    }
```

Comment:
Won't this be inaccurate if it's changed after login, since it is only updated on logged in event?